react/src/components/atoms/buttons/ButtonWithIcon/style.scss

```diff
@@ -45,24 +89,149 @@
     vertical-align: middle;
     fill: $c-gray-dark;
     height: 20px;
-    transition: fill .4s;
+    transition: transform .5s, fill .4s;
   }
 
-  &.ma__icon-green > svg {
-    fill: $c-primary-alt;
+  @each $class, $c in $colors {
+    &.ma__icon--#{$class} > svg {
+      fill: $c;
+    }
   }
 
   &.ma__icon-small > svg {
     height: 10px;
   }
 
+  &--disabled {
+    opacity: 0.5;
+    cursor: not-allowed;
+  }
+
   &--expanded, &:hover {
-    background-color: $c-primary-alt;
-    border-color: $c-primary-alt;
     color: $c-white;
     svg {
       fill: $c-white;
     }
+    &.ma__button-icon {
+      @include ma-button("normal","solid",$c-primary-alt);
+    }
+    @each $class, $c in $colors {
+      &.ma__button-icon--#{$class} {
+        @include ma-button("normal","solid",$c);
+      }
+    }
+  }
+
+  &--secondary, &--expanded.ma__button-icon--secondary, &:hover.ma__button-icon--secondary {
+    @include ma-button("normal","outline",$c-primary-alt);
+    svg {
+      fill: $c-primary-alt;
+    }
+    &:hover {
+      svg {
+        fill: $c-white;
+      }
+    }
+    @each $class, $c in $colors {
+      &.ma__button-icon--#{$class} {
+        @include ma-button("normal","outline",$c);
+        @include ma-button-icon-secondary($c);
+      }
+    }
+  }
+
+  &--tertiary, &--expanded.ma__button-icon--tertiary, &:hover.ma__button-icon--tertiary {
+    @include ma-button("normal","solid",$c-gray-light);
+    color: $c-black;
+    svg {
+      fill: $c-black;
+    }
+  }
+
+  &--quaternary, &--expanded.ma__button-icon--quaternary, &:hover.ma__button-icon--quaternary {
+    min-height: 30px;
+    background-color: transparent;
+    box-shadow: none;
+    border: none;
+    padding: 0;
+    text-transform: none;
+    border-bottom: 2px solid rgba($c-primary-alt,0.75);
```

Comment:
setting the default here is not neccessary, as in the mixin you already set default by passing `$color: $c-primary-alt`

Reply:
also applied to all usage, don't need to set default color as its already set in mixin

Reply:
@clairesunstudio --> not true here since default theme is c-primary-alt that theme class is not appended so I need to set the default with no theme class to c-primary-alt. Unless we want to change this but it is the same way the base button is doing it (buttons.scss), where with no class it is styled in the theme c-primary.